Blog Post - remove inline style from markdown

barbaraperic opened the pull request:
## Changes
- removed inline styles from markdown files 
- added 3 components (`figure`, `figcaption`, `iframe`) in `MarkdownContent` in order to pass `className` values that correspond to removed inline styles

## Note
- I've noticed that page `blog/what’s-new-with-the-filecoin-plus-notary-election-and-filecoin-day-highlights-from-labweek23` returns 404 - haven't looked much into it, happy to open a new ticket for this

mirhamasala commented:
@barbaraperic We're going to need to remove inline styles from all blog posts. And make sure that whatever needs styling is done through `MarkdownContent`

mirhamasala reviewed (changes requested):
@barbaraperic We're going to need to remove inline styles from all blog posts. And make sure that whatever needs styling is done through MarkdownContent

CharlyMartin commented on src/app/_components/MarkdownContent.tsx:
```diff
@@ -16,6 +16,22 @@ export function MarkdownContent({ children }: MarkdownContentProps) {
         img: ({ src, alt }) => (
           <CustomImage src={src as string} alt={alt as string} />
         ),
+        figure: ({ node, ...props }) => (
+          <figure {...props} className="text-center">
+            {props.children}
+          </figure>
+        ),
+        figcaption: ({ node, ...props }) => (
+          <figcaption
+            {...props}
+            className="overflow-hidden overflow-ellipsis whitespace-nowrap break-words text-left font-sans text-[10px] font-thin"
+          >
+            {props.children}
+          </figcaption>
+        ),
+        iframe: ({ node, ...props }) => (
+          <iframe {...props} className="border-0" />
+        ),
```
Why not add the style of these 3 elements in `tailwind.config.js`, to stay consistent with this existing setup?

In the case of `img` just above, we have to render `CustomImage` (a custom component), so we need this extra configuration. With `figure`, `figcaption` and `iframe`, we're only adding style to the existing HTML markup so we don't need to recreate it. Unless I'm missing something? Does that make sense?

barbaraperic replied:
Reason why I included it is to make it self-contained, so the styles only apply when these elements are used in the markdown context. But since we already have `figcaption` in tailwind, it makes sense to move all of those there.  👍

CharlyMartin replied:
I believe the styles we define under `typography` in the `tailwind.config` are only applied when `className="prose"` is applied, which is the case on `MarkdownContent`. So we're safe on that front :)

https://github.com/tailwindlabs/tailwindcss-typography?tab=readme-ov-file#customizing-the-css

CharlyMartin reviewed (commented):
I would move all the style to `tailwind.config.js`, like you're already doing below :)

```
figcaption: {
  a: {
    color: '#cccccc',
    textDecoration: 'none',
  },
},
```

mirhamasala commented on src/content/blog/what’s-new-with-the-filecoin-plus-notary-election-and-filecoin-day-highlights-from-labweek23.md:
```diff
@@ -84,7 +83,7 @@ FF Co-Founding Officer Clara Tsao discussed why the future of AI and enterprise
 **Crossing the Streams: A Ghostbusters-themed Approach to Scaling the Filecoin Ecosystem**
 
-<div style="text-align: center;">
+<figure>
```
If there's not `figcaption` - There's no need to wrap the `iframe` in such a tag.

mirhamasala commented on tailwind.config.js:
```diff
@@ -63,6 +63,13 @@ module.exports = {
+            figcaption: {
+              a: {
+                color: '#cccccc',
+                textDecoration: 'none',
```
is this necessary? Seems like we're already defining it here:

```
            a: {
              textDecoration: 'none',
              '&:hover': {
                textDecoration: 'underline',
              },
              '&:focus, &:focus-visible': {
                outlineWidth: 2,
                outlineStyle: 'solid',
                outlineColor: theme('colors.brand.100'),
              },
            },
            ```

mirhamasala commented on tailwind.config.js:
```diff
@@ -63,6 +63,13 @@ module.exports = {
+
+            figcaption: {
+              a: {
+                color: '#cccccc',
```
Also do we need to change this? [>>](https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/143/files#r1615770850)

barbaraperic replied:
We don't have `#cccccc` as our brand color?

mirhamasala replied:
Remember that these inline styles were randomly added by the content team. We want to refactor and streamline these to align better with the current design of the site. We can change anything and everything we need.

mirhamasala replied:
Also - why don't we just remove it all together and it inherit the `a` styles we set?

mirhamasala commented:
@barbaraperic @CharlyMartin - The current setup would require the content team to know to wrap an iframe into a `figure` tag whenever using a `figcaption`. We won't be able to get away without any sort of training .. but there is this other option that I wanted to talk to you about. This is a rough example. Let me know what you think.


```
import React from 'react';
import ReactMarkdown from 'react-markdown';
import rehypeRaw from 'rehype-raw';
import { CustomImage } from '@/components/CustomImage';

type MarkdownContentProps = {
  children: string;
};

const extractCaption = (node: any) => {
  const nextSibling = node?.parent?.children?.[node?.parent?.children?.indexOf(node) + 1];
  if (nextSibling && nextSibling.type === 'comment' && nextSibling.value.startsWith('caption:')) {
    return nextSibling.value.replace('caption:', '').trim();
  }
  return null;
};

export function MarkdownContent({ children }: MarkdownContentProps) {
  return (
    <ReactMarkdown
      rehypePlugins={[rehypeRaw]}
      className="prose"
      components={{
        img: ({ node, ...props }) => (
          <CustomImage src={props.src as string} alt={props.alt as string} />
        ),
        iframe: ({ node, ...props }) => {
          const caption = extractCaption(node);
          return caption ? (
            <figure>
              <iframe {...props} />
              <figcaption>{caption}</figcaption>
            </figure>
          ) : (
            <iframe {...props} />
          );
        },
      }}
    >
      {children}
    </ReactMarkdown>
  );
}
```

Used liked this in the markdown:

```
 <iframe
    width="560"
    height="315"
    src="https://www.youtube.com/embed/57Ua4AEWe_g?si=nFb8apksN6EgLB_w"
    title="YouTube video player"
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
    allowfullscreen
  ></iframe>
  <!--caption:This is a caption describing the embedded YouTube video.-->
`;
```

mirhamasala commented on tailwind.config.js:
```diff
@@ -56,13 +56,34 @@ module.exports = {
 
             iframe: {
               width: '100%',
+              borderWidth: '0',
```
For all - Or just YouTube?

mirhamasala commented on tailwind.config.js:
```diff
@@ -56,13 +56,34 @@ module.exports = {
+              whiteSpace: 'nowrap',
+              textOverflow: 'ellipsis',
+              textAlign: 'left',
+              fontFamily:
```
Let's not change the `fontFamily`

barbaraperic commented:
@mirhamasala @CharlyMartin let me know what you think about the new video component for Decap. There's a test blog post named "This is a test" where you can test how it works. 


https://github.com/FilecoinFoundationWeb/filecoin-foundation/assets/50910606/7a945a40-8a19-4b13-867f-2728fe8f74d8

barbaraperic commented:
> @mirhamasala @CharlyMartin let me know what you think about the new video component for Decap. There's a test blog post named "This is a test" where you can test how it works.
> 
>  CleanShot.2024-05-28.at.12.24.49.mp4

Sorry realised that we might want to have figcaptions as links, I'll amend.

mirhamasala commented:
>I've noticed that page blog/what’s-new-with-the-filecoin-plus-notary-election-and-filecoin-day-highlights-from-labweek23 returns 404 - haven't looked much into it, happy to open a new ticket for this

@barbaraperic I've fixed all the slugs with an `’` in them in #164  - That was the reason they were breaking.

CharlyMartin commented:
@barbaraperic I love the work you did on the video template, super cool 👏 

I've refactored it a bit to make it clearer, to me at least, with this commit https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/143/commits/8fbc73c346ce156f56a082954fa8c39555806fc4. Does it make sense to you too?

Also two things:
1. The `required: true` validation on `Embedded Video URL` doesn't work, meaning I can publish the article with an empty video URL. So I included this edge case in `renderVideoComponent`
2. I moved the `captionTexts` above the `captionLinks` as I believe it makes more sense for users. You can have texts without links but not the other way around, right?

barbaraperic commented:
> @barbaraperic I love the work you did on the video template, super cool 👏
> 
> I've refactored it a bit to make it clearer, to me at least, with this commit [8fbc73c](https://github.com/FilecoinFoundationWeb/filecoin-foundation/commit/8fbc73c346ce156f56a082954fa8c39555806fc4). Does it make sense to you too?
> 
> Also two things:
> 
> 1. The `required: true` validation on `Embedded Video URL` doesn't work, meaning I can publish the article with an empty video URL. So I included this edge case in `renderVideoComponent`
> 2. I moved the `captionTexts` above the `captionLinks` as I believe it makes more sense for users. You can have texts without links but not the other way around, right?

Love the refactor! Works well for me 👍

mirhamasala commented on tailwind.config.js:
```diff
@@ -63,6 +64,20 @@ module.exports = {
+              wordBreak: 'normal',
+              overflow: 'hidden',
+              whiteSpace: 'nowrap',
+              textOverflow: 'ellipsis',
```
Why do we use an ellipsis to hide the `figcaption` partially and is there a way to see it completely?

barbaraperic replied:
If we don't want the size of the container to cut the paragraph, we can remove the `whiteSpace` property. Otherwise it's truncated and `ellipsis` property will add "...". Perhaps the better option would be, as you suggested, to have it fully readable across all screen devices.

mirhamasala reviewed (approved):
Stellar @barbaraperic 🙌🏼🔥 - LGTM!
